src/code/z_sub_s.c

```diff
@@ -214,17 +213,260 @@ void SubS_UpdateFlags(u16* flags, u16 setBits, u16 unsetBits) {
     *flags = (*flags & ~unsetBits) | setBits;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013AF00.s")
+/**
+ * Fills the weightArray to be used with time paths
+ *
+ * This default weightArray is just an array of the point indicies as floats,
+ * offset to start at the index of the interpolation order minus 1.
+ *
+ * @param[out] weightArray an array of values that are used to compute the progress and the individual weights
+ * @param[in] order the order of the interpolation i.e. the number of points in the interpolation
+ * @param[in] numPoints the number of points to fill, generally the path count + order
+ *
+ * @note `SubS_TimePathing_Update` assumes order to be 3, see SUBS_TIME_PATHING_ORDER
+ */
+void SubS_TimePathing_FillWeightArray(f32 weightArray[], s32 order, s32 numPoints) {
+    s32 i;
+    f32 val = 0.0f;
+
+    for (i = 0; i < numPoints; i++) {
+        if ((i >= order) && (i < (numPoints - order + 1))) {
+            val += 1.0f;
+        }
+        weightArray[i] = val;
+    }
+}
+
+/**
+ * Computes the progress to be used with time paths
+ *
+ * @param[out] progress the progress along the path, used to compute the weights
+ * @param[in] elapsedTime how much time has passed
+ * @param[in] waypointTime how much time per each waypoint
+ * @param[in] totalTime how much time the path should take to travel
+ * @param[in] pathCount the path count
+ * @param[in] order the order of the interpolation i.e. the number of points in the interpolation
+ * @param[in] weightArray see `SubS_TimePathing_FillWeightArray`
+ *
+ * @return s32 0 for error, 1 if still on the path, and 2 if the end of the path should be reached
```

Comment:
Since this function actually doesn't move the actor there, it makes more sense to me as future tense rather than past tense. Though don't really have a strong opinion either way.